client/my-sites/domains/domain-management/site-redirect/index.jsx

```diff
@@ -3,151 +3,222 @@
  */
 import React from 'react';
 import classNames from 'classnames';
+import { connect } from 'react-redux';
 import page from 'page';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-import analyticsMixin from 'lib/mixins/analytics';
 import Header from 'my-sites/domains/domain-management/components/header';
 import FormButton from 'components/forms/form-button';
 import FormFieldset from 'components/forms/form-fieldset';
 import FormFooter from 'my-sites/domains/domain-management/components/form-footer';
 import FormLabel from 'components/forms/form-label';
 import FormTextInputWithAffixes from 'components/forms/form-text-input-with-affixes';
 import Main from 'components/main';
-import SiteRedirectNotice from './notice';
+import Notice from 'components/notice';
+import notices from 'notices';
 import paths from 'my-sites/domains/paths';
 import * as upgradesActions from 'lib/upgrades/actions';
 import Card from 'components/card/compact';
 import SectionHeader from 'components/section-header';
-
-const SiteRedirect = React.createClass( {
-	mixins: [ analyticsMixin( 'domainManagement', 'siteRedirect' ) ],
-
-	propTypes: {
+import {
+	composeAnalytics,
+	recordGoogleEvent,
+	recordTracksEvent,
+} from 'state/analytics/actions';
+
+class SiteRedirect extends React.Component {
+	static propTypes = {
 		location: React.PropTypes.object.isRequired,
 		selectedDomainName: React.PropTypes.string.isRequired,
 		selectedSite: React.PropTypes.oneOfType( [
 			React.PropTypes.object,
 			React.PropTypes.bool
 		] ).isRequired
-	},
+	};
+
+	state = {
+		location: this.props.location.value
+	};
 
 	componentWillMount() {
 		upgradesActions.fetchSiteRedirect( this.props.selectedSite.domain );
-	},
+	}
 
-	componentWillReceiveProps: function( nextProps ) {
+	componentWillReceiveProps( nextProps ) {
 		if ( this.props.location.value !== nextProps.location.value ) {
 			this.setState( {
 				location: nextProps.location.value
 			} );
 		}
-	},
+	}
 
 	componentWillUnmount() {
-		upgradesActions.closeSiteRedirectNotice( this.props.selectedSite.domain );
-	},
+		this.closeRedirectNotice();
+	}
 
-	getInitialState() {
-		return {
-			location: this.props.location.value
-		};
-	},
+	closeRedirectNotice = () => {
+		upgradesActions.closeSiteRedirectNotice( this.props.selectedSite.domain );
+	};
 
-	handleChange( event ) {
+	handleChange = ( event ) => {
 		let location = event.target.value;
 
 		// Removes the protocol part
 		location = location.replace( /.*:\/\//, '' );
```

Comment:
My only problem is that in theory the redirect could be to any protocol. Gah, maybe we should make `stripProtocol/withoutProtocol`?